File: CleanJavaRx/app/src/main/java/com/vascome/fogtail/data/thread/DataThread.java
package com.vascome.fogtail.data.thread;
import com.vascome.fogtail.presentation.base.domain.executor.ExecutionThread;

import javax.inject.Inject;
import javax.inject.Singleton;

import io.reactivex.Scheduler;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

/**
 * MainThread (UI Thread) implementation based on a {@link Scheduler}
 * which will execute actions on the Android UI thread
 */
@Singleton
public class DataThread implements ExecutionThread {

    @Inject
    DataThread() {}

    @Override
    public Scheduler getUIScheduler() {
        return AndroidSchedulers.mainThread();
    }

    @Override
    public Scheduler getIOScheduler() {
        return Schedulers.io();
    }
}

File: CleanKotlin/app/src/main/java/com/vascome/fogtail/data/thread/DataThread.kt
package com.vascome.fogtail.data.thread

import com.vascome.fogtail.presentation.base.domain.executor.ExecutionThread

import javax.inject.Inject
import javax.inject.Singleton

import io.reactivex.Scheduler
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers

/**
 * MainThread (UI Thread) implementation based on a [Scheduler]
 * which will execute actions on the Android UI thread
 */
@Singleton
class DataThread
@Inject internal constructor() : ExecutionThread {

    override val uiScheduler: Scheduler
        get() = AndroidSchedulers.mainThread()

    override val ioScheduler: Scheduler
        get() = Schedulers.io()
}
